### DISCUSSION TITLE
[Nuxt 3]Available to set files to be excluded from `pages` directory

### DISCUSSION THREAD

**Discussion by yaquawa** [nuxt/nuxt#17980] [Ideas]
Just like https://github.com/hannoeru/vite-plugin-pages#exclude , have the ability to set files to be excluded from auto generating routes would be useful.

I'd like to put partial components of page in the `pages` directory but not in the `components` directory like:

* pages
  * dashboard
  * index.vue
    * `__partial__`
      * Header.vue
      * Body.vue

These vue files specific to the pages aren't really globally reusable "components", so I'd like to keep them together within the pages directory. So any file under `__partial__` should not be any routes that will be automatically generated.

**Comment by danielroe** [comment]
Let's track in https://github.com/nuxt/framework/issues/2998.
